src/reason-parser/reason_pprint_ast.ml

```diff
@@ -7550,15 +7555,15 @@ let printer = object(self:'self)
             s
         in
         makeList
-          ~break:Layout.Always_rec
+          ~break:(if List.length s = 0 then Layout.IfNeed else Layout.Always_rec)
```

Comment:
I have seen this logic in more than a few places, maybe we could try to do this check inside makeList and see the snapshots?

Reply:
Apologies for the late reply. I've added this to the changes of this PR.